src/Components/Aspire.Qdrant/AspireQdrantExtensions.cs

```diff
@@ -0,0 +1,97 @@
+    /// <exception cref="InvalidOperationException">If required ConnectionString is not provided in configuration section</exception>
+    public static void AddQdrantClient(
+        this IHostApplicationBuilder builder,
+        string connectionName,
+        Action<QdrantSettings>? configureSettings = null)
+    {
+        AddQdrant(builder, DefaultConfigSectionName, configureSettings, connectionName, serviceKey: null);
+    }
+
+    /// <summary>
+    /// Registers <see cref="QdrantClient" /> as a singleton for given <paramref name="name" /> in the services provided by the <paramref name="builder"/>.
+    /// Configures logging and telemetry for the <see cref="QdrantClient" />.
+    /// </summary>
+    /// <param name="builder">The <see cref="IHostApplicationBuilder" /> to read config from and add services to.</param>
+    /// <param name="name">The connection name to use to find a connection string.</param>
+    /// <param name="configureSettings">An optional method that can be used for customizing the <see cref="QdrantSettings"/>. It's invoked after the settings are read from the configuration.</param>
+    /// <remarks>Reads the configuration from "Aspire:Qdrant:Client" section.</remarks>
+    /// <exception cref="InvalidOperationException">If required ConnectionString is not provided in configuration section</exception>
+    public static void AddKeyedQdrantClient(
+        this IHostApplicationBuilder builder,
+        string name,
+        Action<QdrantSettings>? configureSettings = null)
+    {
+        AddQdrant(builder, $"{DefaultConfigSectionName}:{name}", configureSettings, connectionName: name, serviceKey: name);
+    }
+
+    private static void AddQdrant(
+        this IHostApplicationBuilder builder,
+        string configurationSectionName,
+        Action<QdrantSettings>? configureSettings,
+        string connectionName,
+        string? serviceKey)
+    {
+        ArgumentNullException.ThrowIfNull(builder);
+
+        var settings = new QdrantSettings();
+        builder.Configuration.GetSection(configurationSectionName).Bind(settings);
+
+        if (builder.Configuration.GetConnectionString(connectionName) is string connectionString)
+        {
+            settings.ParseConnectionString(connectionString);
+        }
+
+        if (builder.Configuration[$"{DefaultConfigSectionName}:{connectionName}:Key"] is string apiKey)
+        {
+            settings.Key = apiKey;
+        }
+
+        configureSettings?.Invoke(settings);
+
+        if (serviceKey is null)
+        {
+            builder.Services.AddSingleton(_ => ConfigureQdrant());
+        }
+        else
+        {
+            builder.Services.AddKeyedSingleton(serviceKey, (sp, key) => ConfigureQdrant());
+        }
+
+        QdrantClient ConfigureQdrant()
+        {
+            if (settings.Endpoint is not null)
+            {
+                return new QdrantClient(settings.Endpoint, apiKey: settings.Key);
```

Comment:
Is there any sort of logging, tracing, metrics, health checks that can be enabled?

Reply:
On the client, not yet. I was looking to actually wire up the metrics for the server _via_ the client, but not doing that ;-).

Shay is hopefully going to get some time to work on adding Tracing but it's just not in yet. Tracking: https://github.com/qdrant/qdrant-dotnet/issues/49

Reply:
It looks like Logging is supported.

https://github.com/qdrant/qdrant-dotnet/blob/b3b80fddbb82bc36c9de9ea7100da93b3624c742/src/Qdrant.Client/QdrantClient.cs#L59-L63